qiskit/optimization/algorithms/grover_optimizer.py

```diff
@@ -395,13 +373,36 @@ def n_output_qubits(self) -> int:
         """
         return self._n_output_qubits
 
+    @property
+    def problem(self) -> QuadraticProgram:
```

Comment:
I get why this is being passed in here, to provision the func_dict, but it seems so generic that such a field would almost seem to more (be passed into and) belong to an OptimizationResult. I get that its needed for func_dict, I am just wondering more about storing it and exposing it as a field here when it seems much more generic across Optimizers to include the problem the solve in the result - if we really want to do that.

Reply:
I just talked with @stefan-woerner and we can also just remove the func_dict entirely, then we also don't need to store the quadratic program 🙂